### PR TITLE
Add `-minimum-slang-optimization` to favor compile time.

### PR DESCRIPTION
**Pull request by csyonghe**
## New Feature

- Add `-minimum-slang-optimization` compiler option to enable a minimum set of optimizations in the Slang compiler. This speeds up the compilation time in Slang for users who wish to favor faster compile times.

### CODE REVIEW

File: slang.h [hunk 1]
```diff
@@ -862,6 +862,7 @@ extern "C"
             SourceEmbedName,
             SourceEmbedLanguage,
             DisableShortCircuit,   // bool
+            MinimumSlangOptimization, // bool
```
Comment (kaizhangNV) [inline slang.h:865]: I'm wondering that if we should use -O<n> to indicate the level of optimization?

Comment (csyonghe) [reply slang.h:865]: For now I don't want to mix these two things up just yet. This allows `-O` to continue to affect downstream compilation without being tied to slang's logic.

**Review by jkwak-work — APPROVED** [review]
Looks good to me.

I wonder how faster it can be with the new option.

**Comment by jkwak-work** [comment]
Probably it is expected but I noticed that some of slang-test tests are failing when the option is enabled.

**Comment by csyonghe** [comment]
@jkwak-work this is fine. We have tests that assumes the optimizations exist and is looking for a specific pattern that is no longer the case without the optimization.

**Pull request merged**